aten/src/ATen/native/cpu/IndexKernel.cpp

```diff
@@ -569,6 +569,145 @@ void cpu_vflip_memcpy(at::TensorIterator& iter) {
   iter.cast_outputs();
 }
 
+std::array<char, 32> generate_vec_hflip_reg_mask(int64_t data_stride) {
```

Comment:
Do you still plan to add AVX-512 support?

Reply:
Yes, but later. Final version with avx2 and avx512 supported should refactor all these methods and probably do things a bit differently.